lib/remote.js

```diff
@@ -5,7 +5,8 @@ const pacoteVersion = require('../package.json').version
 const fetch = require('npm-registry-fetch')
 const Minipass = require('minipass')
 // The default registry URL is a string of great magic.
-const magic = /^https?:\/\/registry\.npmjs\.org\//
+const magicHost = /^https?:\/\/registry\.npmjs\.org\//
```

Comment:
While we're in here we should take the opportunity to not use regex and instead let `new URL()` do what it does best.

Reply:
Agreed!